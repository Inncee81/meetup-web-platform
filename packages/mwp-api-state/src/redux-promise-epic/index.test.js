import { makeCallEpics } from './index';

describe('makeCallEpics', () => {
	test('every epic calls store.dipatch', () => {
		const store = { dispatch: jest.fn() };
		const epics = [
			() => Promise.resolve(['action1']),
			() => Promise.resolve(['action2']),
			() => Promise.resolve(['action3']),
		];

		const callEpics = makeCallEpics(...epics);
		return Promise.all(callEpics('foo', store)).then(() => {
			expect(store.dispatch.mock.calls).toMatchSnapshot();
		});
	});
	test('delayed actions arrive later', () => {
		const store = { dispatch: jest.fn() };
		const delayedAction = 'action0';
		const delayedEpic = () =>
			new Promise(resolve => setTimeout(() => resolve([delayedAction]), 5));
		const epics = [
			delayedEpic,
			() => Promise.resolve(['action1']),
			() => Promise.resolve(['action2']),
			() => Promise.resolve(['action3']),
		];

		const callEpics = makeCallEpics(...epics);
		return Promise.all(callEpics('foo', store)).then(() => {
			expect(store.dispatch).toHaveBeenLastCalledWith(delayedAction);
		});
	});
});
