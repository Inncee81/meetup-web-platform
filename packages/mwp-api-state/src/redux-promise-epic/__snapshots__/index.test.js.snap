// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeCallEpics every epic calls store.dipatch 1`] = `
Array [
  Array [
    "action1",
  ],
  Array [
    "action2",
  ],
  Array [
    "action3",
  ],
]
`;
